### PR TITLE
[node] 792 onchain transactions

### PR DESCRIPTION
**Pull request by rhlsthrm**
## The Problem
#792 

## The Solution
Make sure onchain transactions of all types (collateral, deposit, withdrawal) are stored in node's DB.

## Checklist:

- [ ] I am making this PR against staging, not master
- [ ] My code follows the code style of this project.
- [ ] I have described any backwards-incompatibility implications above.
- [ ] I have highlighted which parts of the code should be reviewed most carefully.
- [ ] If my change requires a change to the documentation, I have updated it accordingly.

### CODE REVIEW

File: modules/cf-core/src/methods/app-instance/uninstall/controller.ts [hunk 1]
```diff
@@ -20,31 +20,30 @@ export default class UninstallController extends NodeController {
...
     requestHandler: RequestHandler,
-    params: CFCoreTypes.UninstallVirtualParams
+    params: CFCoreTypes.UninstallVirtualParams,
   ): Promise<string[]> {
     const { store } = requestHandler;
     const { appInstanceId } = params;
 
     const sc = await store.getChannelFromAppInstanceID(appInstanceId);
 
-    if (sc.freeBalance.identityHash === appInstanceId) {
```
Comment (LayneHaber) [inline modules/cf-core/src/methods/app-instance/uninstall/controller.ts]: why did this get removed? still seems valid

File: modules/cf-core/src/methods/app-instance/uninstall/controller.ts [hunk 2]
```diff
@@ -20,31 +20,30 @@ export default class UninstallController extends NodeController {
...
     if (!appInstanceId) {
       throw Error(NO_APP_INSTANCE_ID_TO_UNINSTALL);
     }
 
+    const sc = await store.getChannelFromAppInstanceID(appInstanceId);
+
+    if (sc.freeBalance.identityHash === appInstanceId) {
+      throw Error(CANNOT_UNINSTALL_FREE_BALANCE(sc.multisigAddress));
+    }
+
```
Comment (LayneHaber) [inline modules/cf-core/src/methods/app-instance/uninstall/controller.ts]: jk i see it was just moved, which makes more sense

File: modules/node/src/cfCore/cfCore.service.ts [hunk 3]
```diff
@@ -154,9 +146,40 @@ export class CFCoreService {
...
+    logger.debug(
+      `Calling ${ProtocolTypes.chan_withdraw} with params: ${stringify({
+        amount,
+        multisigAddress,
+        tokenAddress: assetId,
+      })}`,
+    );
+    const withdrawRes = await this.cfCore.rpcRouter.dispatch({
+      id: Date.now(),
+      methodName: ProtocolTypes.chan_deposit,
```
Comment (LayneHaber) [inline modules/node/src/cfCore/cfCore.service.ts]: shouldn't it be `chan_withdraw`

File: modules/node/src/channel/channel.service.spec.ts [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+      appDefinitionAddress: mkAddress("0xa"),
+      chainId: 1337,
+      name: "SimpleLinkedTransferApp",
+      outcomeType: OutcomeType.SINGLE_ASSET_TWO_PARTY_COIN_TRANSFER,
+      stateEncoding: "",
+    };
+  }
+}
+
+describe.skip("Channel Service", () => {
```
Comment (LayneHaber) [inline modules/node/src/channel/channel.service.spec.ts]: this should be unskipped before merging

File: modules/node/src/channel/channel.service.spec.ts [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+        { provide: getRepositoryToken(OnchainTransaction), useClass: MockOnchainTransactionRepository },
+      ],
+    }).compile();
+    channelService = moduleRef.get<ChannelService>(ChannelService);
+    console.log("channelService: ", channelService);
+  });
+
+  it("should add deposits to the onchain transaction table", async () => {
+    const result = await channelService.deposit(mkAddress("0xAAA"), One);
+  });
```
Comment (LayneHaber) [inline modules/node/src/channel/channel.service.spec.ts]: could probably use some more `expect`s here

File: modules/test-runner/src/util/helpers/requestDepositRights.ts [hunk 6]
```diff
@@ -64,6 +62,7 @@ export const requestDepositRights = async (
       responderDepositTokenAddress: assetId,
       timeout: Zero,
     };
+    console.log("HEREEEEEEE");
```
Comment (LayneHaber) [inline modules/test-runner/src/util/helpers/requestDepositRights.ts]: lol we can probably remove this

File: modules/node/src/cfCore/cfCore.service.ts (hunk 3 above)
Comment (rhlsthrm) [reply modules/node/src/cfCore/cfCore.service.ts]: Clearly this has not been tested.

File: modules/node/src/channel/channel.service.spec.ts (hunk 4 above)
Comment (rhlsthrm) [reply modules/node/src/channel/channel.service.spec.ts]: This isn't a real test, I was playing around with the test framework a bit.

File: modules/node/src/channel/channel.service.spec.ts (hunk 5 above)
Comment (rhlsthrm) [reply modules/node/src/channel/channel.service.spec.ts]: Not a real test, the real test is in the integration tests.

**Pull request merged**